src/main/java/hudson/plugins/ircbot/v2/IRCColorizer.java

```diff
@@ -12,23 +20,150 @@
  * 
  * @author syl20bnr
  * @author kutzi
+ * @author tao 
+ * @author hongjae
+ * @author austin
  */
 // See http://flylib.com/books/en/4.74.1.47/1/ for some tips on IRC colors
 public class IRCColorizer {
+	
+	private static final Logger LOGGER = Logger.getLogger(IRCColorizer.class.getName());
     
     /**
      * Very simple pattern to recognize test results.
      */
     private static final Pattern TEST_CLASS_PATTERN = Pattern.compile(".*test.*", Pattern.CASE_INSENSITIVE);
-
+    
+    //userPattern store the String/Pattern->color preference for each specific user
+    //e.g.userPattern[nickname][user_prefered_string_pattern]
+    private static HashMap<String, HashMap<String, String> > userPattern; 
+    private static HashMap<String, HashMap<ResultTrend, String> > themes; 
+    private static String currentTheme;
+    private static ArrayList<ResultTrend> result;
```

Comment:
Common coding best practices say to prefer interfaces over concrete implementation.
So please declare these as Map resp. List